include/simdjson/generic/ondemand/document-inl.h

```diff
@@ -134,6 +134,17 @@ simdjson_really_inline simdjson_result<std::string_view> document::raw_json_toke
   return std::string_view(reinterpret_cast<const char*>(_iter.peek_start()), _iter.peek_start_length());
 }
 
+simdjson_really_inline simdjson_result<value> document::at_pointer(std::string_view json_pointer) noexcept {
+  switch ((*this).type()) {
```

Comment:
@lemire 
The issue seems to by caused by the fact that `type()` returns *simdjson_result<json_type>* which invalidates the switch statement and I need to convert it to *json_type*. I tried a few things but I am stuck.

Reply:
```c++
json_type t;
SIMDJSON_TRY(type().get(t));
switch (t) {
...
```

Should do the trick.

Error code style is intrinsically verbose, unfortunately.

Reply:
Just to explain @jkeiser 's proposal. What `SIMDJSON_TRY` does for you is this:

```C++
#define SIMDJSON_TRY(EXPR) { auto _err = (EXPR); if (_err) { return _err; } }
```

So you could write it the long way as...

```C++
json_type t;
{
auto error = type().get(t);
if(error){ return error; }
}
switch (t) {
```

In C++20, you could do...

```C++
if(auto error = type().get(t); error) { return error;}
```


That's the general pattern in simdjson. Call get, check the error, if the error is "ON", then return it, otherwise continue.

It is slightly 'verbose' in the same that you need the variable declaration line...

```C++
json_type t;
```

But I don't think that it is a fault.